### PR TITLE
Adding basepath/url settings to the installer

### LINKED ISSUE #6720: Base URL and Base Path shouldn't be guessed on every single request
On the heel of #6679, I'd like to move the App auto-detection of the base path and base URL to the install phase. These values then could be written in the config file skeleton and not be detected/re-written on each call.

Not only it is useless for correctly configured nodes, but it can produce spectacular failures when something goes wrong.

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixes #6720 
FollowUp #6989 

Tested with local node and some wizard/automatic installations.

This is the new screen:
grafik

What's new:
- url, urlpath, basepath, ssl_policy and hostname is now set during install
- wizard contains the new screen during installation for the settings above
- the automatic install now needs at least the `--url` parameter for setting the right hostname (isn't detectable because lack of  `$_SERVER` variable) - or a given config file like before

I think I'm done with baseurl/basepath :D

### CODE REVIEW

File: src/Core/Installer.php [hunk 1]
```diff
@@ -243,7 +248,7 @@ public function checkPHP($phppath = null, $required = false)
 			$help .= EOL . EOL;
 			$tpl = Renderer::getMarkupTemplate('field_input.tpl');
 			$help .= Renderer::replaceMacros($tpl, [
-				'$field' => ['phpath', L10n::t('PHP executable path'), $phppath, L10n::t('Enter full path to php executable. You can leave this blank to continue the installation.')],
+				'$field' => ['config.php_path', L10n::t('PHP executable path'), $phppath, L10n::t('Enter full path to php executable. You can leave this blank to continue the installation.')],
```
Comment (MrPetovan) [inline src/Core/Installer.php]: This isn't consistent with the input fields in `view/templates/install/*` which are named `config-php_path`. Is this intended? Am I comparing apples to oranges?

Comment (nupplaphil) [reply src/Core/Installer.php]: I'll change it and leave a reminder that we mix backend classes with the "presentation layer" (code smell)

Thanks for the hint

File: src/Module/Install.php [hunk 2]
```diff
@@ -44,22 +50,23 @@ public static function init()
 	{
 		$a = self::getApp();
 
-		if (!$a->getMode()->isInstall()) {
```
[the remark this answers is not in the document]
Comment (nupplaphil) [reply src/Module/Install.php]: Good question .. seems like a rebase issue.. Thanks!

**Comment by nupplaphil** [comment]
Found a rebase regression (missing username during wizard)

**Comment by nupplaphil** [comment]
Wizard works again, auto-install too

**Pull request merged**